src/dll.py

```diff
@@ -0,0 +1,90 @@
+"""Implementation of Doubly-Linked list with a head and tail."""
+from linked_list import LinkedList
+from linked_list import Node
+
+
+class Dll(object):
+    """Doubly-Linked List class object."""
+
+    def __init__(self):
+        """Doubly-linked list initialization.
+
+        Composed of some attributes from linked-list, and also has a tail.
+        """
+        self._linkedlist = LinkedList()
+        self.head = self._linkedlist.head
+        self._length = self._linkedlist._length
+        self.tail = None
+
+    def push(self, data):
+        """Push node to head of list."""
+        prev_head = self.head
+        new_head = self._linkedlist.push(data)
+        if self.tail is None:
+            self.tail = new_head
+        if self.head:
+            prev_head.prev = new_head
+        self.head = new_head
+        self.head.next_node = prev_head
+        self._length += 1
+        self.head.prev = None
+
+    def pop(self):
+        """Remove node at head of list."""
+        if not self.head:
+            raise IndexError('List empty')
+        deleted_node = self.head.data
+        self._length -= 1
+        if not self.head.next_node:
+            self.head = None
+            self.tail = None
+        else:
+            self.head = self.head.next_node
+            self.head.prev = None
+        return deleted_node
+
+    def append(self, data):
+        """Append method for Dll to add to tail."""
+        prev_tail = self.tail
+        new_tail = Node(data)
```

Comment:
You're using a Node object that does not inherently have a `prev` pointer, and are forcing one on later. Would be better to just make a Node object that has access to its `prev` pointer.